lib/board.rb

```diff
@@ -0,0 +1,62 @@
+class Board
+  attr_accessor :cells
+
+
+
+  def initialize
+    self.reset!
+
+  end
+
+  def reset!
+    @cells = [" "," "," "," "," "," "," "," "," "]
+  end
+
+  def display
+    b = @cells
+    puts " #{b[0]} | #{b[1]} | #{b[2]} "
+    puts "-----------"
+    puts " #{b[3]} | #{b[4]} | #{b[5]} "
+    puts "-----------"
+    puts " #{b[6]} | #{b[7]} | #{b[8]} "
+  end
+
+  def position(pos)
+    @cells[pos.to_i-1]
+  end
+
+  def full?
+    @cells.all? do |pos|
+      pos == "X" || pos == "O"
+    end
+  end
+
+  def turn_count
+    count = 0
+    @cells.each do |pos|
+      count +=1 if (pos != " ")
+    end
+    count
+  end
+
+  def taken?(pos)
+    @cells[pos.to_i-1] != " "
+  end
+
+  def valid_move?(pos)
```

Comment:
Use 'and' operator (`&&`) to combine multiple logical expressions and utilize their implicit return value.
```
input.to_i.between?(1,9) && !taken(input)
```